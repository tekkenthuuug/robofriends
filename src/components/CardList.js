import React from 'react';
import Card from './Card';

const CardList = ({ robots }) => {
	if (true) {
		throw new Error("FUCK FUCK FUCK FUCK");
	}
	return (
		<div>
			{
				robots.map((user, i) => {
					return (
						<Card
							key={i}
							id={robots[i].id}
							name={robots[i].name}
							email={robots[i].email}
						/>
					);
				})
			}
		</div>
	);
}

export default CardList;