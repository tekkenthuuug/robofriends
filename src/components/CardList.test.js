import { shallow } from "enzyme";
import React from "react";
import CardList from "./CardList.js";

it("expect to render CardList component", () => {
	const mockRobots = [
		{
			id       : 1,
			name     : "John Snow",
			username : "JohnJohn",
			email    : "john@gmail.com"
		},
		{
			id       : 2,
			name     : "Darek Snow",
			username : "Daks",
			email    : "darek@gmail.com"
		}
	];
	expect(shallow(<CardList robots={mockRobots} />)).toMatchSnapshot();
});
