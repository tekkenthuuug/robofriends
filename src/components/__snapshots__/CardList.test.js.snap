// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render CardList component 1`] = `
<div>
  <Card
    email="john@gmail.com"
    id={1}
    key="0"
    name="John Snow"
  />
  <Card
    email="darek@gmail.com"
    id={2}
    key="1"
    name="Darek Snow"
  />
</div>
`;
